src/core/Config.cpp

```diff
@@ -476,19 +476,21 @@ void Config::init(const QString& configFileName, const QString& localConfigFileN
     // Upgrade from previous KeePassXC version which stores its config
     // in ~/.cache on Linux instead of ~/.local/state.
     // Move file to correct location before continuing.
-    QString oldLocalConfigPath;
-    oldLocalConfigPath = QStandardPaths::writableLocation(QStandardPaths::GenericCacheLocation) + "/keepassxc";
-    QString suffix;
+    if (!QFile::exists(localConfigFileName)) {
+        QString oldLocalConfigPath;
+        oldLocalConfigPath = QStandardPaths::writableLocation(QStandardPaths::GenericCacheLocation) + "/keepassxc";
```

Comment:
You should not separate declaration/definition and assignment. Rest looks good.

Reply:
Fixed in https://github.com/keepassxreboot/keepassxc/pull/9755/commits/85a2b740ffd2423a30d90dd9e31e2d77ac3f54aa